chainer/links/connection/mgu.py

```diff
@@ -0,0 +1,66 @@
+import numpy
+
+import chainer
+from chainer.functions.activation import sigmoid
+from chainer.functions.activation import tanh
+from chainer.functions.array import concat
+from chainer.functions.math import linear_interpolate
+from chainer import link
+from chainer.links.connection import linear
+
+
+class MGU(link.Chain):
+
+    def __init__(self, n_inputs, n_units):
+        super(MGU, self).__init__(
+            W_f=linear.Linear(n_inputs + n_units, n_units),
+            W_h=linear.Linear(n_inputs + n_units, n_units)
+        )
+
+    def __call__(self, h, x):
+        f = sigmoid.sigmoid(self.W_f(concat.concat([h, x])))
+        h_bar = tanh.tanh(self.W_h(concat.concat([f * h, x])))
+        h_new = linear_interpolate.linear_interpolate(f, h_bar, h)
+        return h_new
+
+
+class StatefulMGU(MGU):
```

Comment:
It have the same problem as one in SGD and DSGU in #1115. `StatefulMGU` may not be the subclass of `MGU`.